src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -485,11 +485,17 @@ bool Unit::IsWithinMeleeRange(Unit const* obj) const
     float dz = GetPositionZMinusOffset() - obj->GetPositionZMinusOffset();
     float distsq = dx*dx + dy*dy + dz*dz;
 
-    float maxdist = GetCombatReach() + obj->GetCombatReach() + 4.0f / 3.0f;
+    float maxdist = GetMeleeRange(obj);
 
     return distsq <= maxdist * maxdist;
 }
 
+float Unit::GetMeleeRange(Unit const* target) const
+{
+    float range = this->GetCombatReach() + target->GetCombatReach() + 4.0f / 3.0f;
```

Comment:
yeah I agree with @killradio. Even if I check if target is null, I don't see how I could handle that case anyway. This function shouldn't be called with NULL. And it shouldn't be called in AI scripts anyway. Only by the core.